test/integration/dialects/postgres/dao.test.js

```diff
@@ -393,6 +393,142 @@ if (dialect.match(/^postgres/)) {
           expect(enums[0].enum_value).to.equal("{neutral,happy,sad,ecstatic,meh,joyful}");
         });
       });
+
+      describe('ARRAY(ENUM)', function () {
+        it('should be able to ignore enum types that already exist', function() {
+          var User = this.sequelize.define('UserEnums', {
+            permissions: DataTypes.ARRAY(DataTypes.ENUM([
+              'access',
+              'write',
+              'check',
+              'delete'
+            ]))
+          });
+
+          return User.sync({ force: true }).then(function() {
+            return User.sync();
+          });
+        });
+
+        it('should be able to create/drop enums multiple times', function() {
+          var User = this.sequelize.define('UserEnums', {
+            permissions: DataTypes.ARRAY(DataTypes.ENUM([
+              'access',
+              'write',
+              'check',
+              'delete'
+            ]))
+          });
+
+          return User.sync({ force: true }).then(function() {
+            return User.sync({ force: true });
+          });
+        });
+
+        it('should be able to add values to enum types', function() {
+          var User = this.sequelize.define('UserEnums', {
+            permissions: DataTypes.ARRAY(DataTypes.ENUM([
+              'access',
+              'write',
+              'check',
+              'delete'
+            ]))
+          });
+
+          return User.sync({ force: true }).bind(this).then(function() {
+            User = this.sequelize.define('UserEnums', {
+              permissions: DataTypes.ARRAY(
+                DataTypes.ENUM('view', 'access', 'edit', 'write', 'check', 'delete')
+              )
+            });
+
+            return User.sync();
+          }).then(function() {
+            return this.sequelize.getQueryInterface().pgListEnums(User.getTableName());
+          }).then(function (enums) {
+            expect(enums).to.have.length(1);
+            expect(enums[0].enum_value).to.equal("{view,access,edit,write,check,delete}");
```

Comment:
I don't know if this order would ever be different? Maybe do a substr + split and assert on the array of items

Reply:
Order wont change. I ensure you this wont turn into an intermittent failing test :D